gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/javaversions/LazilyConfiguredMapping.java

```diff
@@ -23,25 +23,32 @@
 import java.util.Optional;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.function.Supplier;
+import javax.annotation.concurrent.GuardedBy;
 import org.gradle.api.Action;
 
 final class LazilyConfiguredMapping<K, V, A> {
     private final Supplier<V> valueFactory;
+
+    @GuardedBy("this")
     private final List<LazyValues<K, V, A>> values = new ArrayList<>();
+
+    @GuardedBy("this")
     private final Map<K, Optional<V>> computedValues = new HashMap<>();
```

Comment:
Considered swapping to a ConcurrentHashmap which handles nested mutations more gracefully, however that's not expected to happen (and isn't related to this issue).